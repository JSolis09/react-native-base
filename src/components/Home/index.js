import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { Button, COLOR } from 'react-native-material-ui';
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5';
import {StyleSheet, Text, View} from 'react-native';

const styles = StyleSheet.create({
    container: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      backgroundColor: '#F5FCFF',
    },
    icons: {
        textAlign: 'center',
        margin: 10,
    },
    icon: {
        marginLeft: 5,
        marginRight: 5,
    },
    welcome: {
      fontSize: 20,
      textAlign: 'center',
      margin: 10,
    },
    instructions: {
      textAlign: 'center',
      color: '#333333',
      marginBottom: 5,
    },
    otherText: {
      textAlign: 'center',
      color: COLOR.yellow900,
    }
  });

export default class Home extends Component {
    static propTypes = {
        message: PropTypes.string,
        goPageOne: PropTypes.func.isRequired
    };

    goPageOne() {
        const { navigate } = this.props.navigation;
        navigate('PageOne');
    }
    
    render() {
        const { message } = this.props;
        return(
            <View style={styles.container}>
                <Text style={styles.icons}>
                    <FontAwesome5 style={styles.icon} color="#90ee02" size={50} name={'android'} />
                    <FontAwesome5 style={styles.icon} color="#000000" size={50} name={'apple'} />
                </Text>
                <Text style={styles.welcome}>Welcome to React Native!</Text>
                <Text style={styles.instructions}>To get started, edit App.js.</Text>
                <Text style={styles.otherText}>{message}</Text>
                <Button primary text="Go PageOne" onPress={this.goPageOne.bind(this)} />
            </View>
        );
    }
}
