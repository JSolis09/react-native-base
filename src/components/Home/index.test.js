import React from 'react';
import { create } from 'react-test-renderer';
import Home from './index';

describe('Home Component', () => {
    it('should render correctly', () => {
        const props = {
            message: 'Default Message'
        };
        const tree = create(<Home {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});