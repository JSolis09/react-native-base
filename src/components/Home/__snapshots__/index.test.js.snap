// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#90ee02",
            "fontSize": 50,
          },
          Object {
            "marginLeft": 5,
            "marginRight": 5,
          },
          Object {
            "fontFamily": "Font Awesome 5 Brands",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "400",
          },
        ]
      }
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 50,
          },
          Object {
            "marginLeft": 5,
            "marginRight": 5,
          },
          Object {
            "fontFamily": "Font Awesome 5 Brands",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "400",
          },
        ]
      }
    >
      
    </Text>
  </Text>
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to React Native!
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    To get started, edit App.js.
  </Text>
  <Text
    style={
      Object {
        "color": "#f57f17",
        "textAlign": "center",
      }
    }
  >
    Default Message
  </Text>
  <View
    accessible={true}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="box-only"
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 2,
          "flexDirection": "row",
          "height": 36,
          "justifyContent": "center",
          "paddingHorizontal": 16,
        },
        undefined,
        false,
        false,
        false,
        Object {},
        undefined,
      ]
    }
    testID={null}
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 14,
            "fontWeight": "500",
          },
          undefined,
          false,
          false,
          false,
          Object {
            "color": "#2196f3",
          },
          undefined,
        ]
      }
    >
      GO PAGEONE
    </Text>
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.87)",
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
      <View
        pointerEvents="none"
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.87)",
            "borderRadius": 100,
            "height": 200,
            "left": -100,
            "opacity": 0.12,
            "position": "absolute",
            "top": -100,
            "transform": Array [
              Object {
                "scale": 0,
              },
            ],
            "width": 200,
            "zIndex": 1,
          }
        }
      />
    </View>
  </View>
</View>
`;
